langchain/src/vectorstores/hnswlib.ts

```diff
@@ -133,6 +131,8 @@ export class HNSWLib extends SaveableVectorStore {
   }
 
   async save(directory: string) {
+    const fs = await import("node:fs/promises");
```

Comment:
is it clear which methods and classes require node imports?

what happens if I try to call `save` in browser/edge/deno/etc?

Reply:
In deno it would work. In other envs it wouldn't. One thing I haven't done yet is add some indication in the docs of which methods are node-only, that's planned